frontend/src/app/sign-up/page.tsx

```diff
@@ -50,32 +50,31 @@ export default function Home() {
       return;
     }
 
-    // mock for backend
-    const res = await fetch("https://jsonplaceholder.typicode.com/user", {
-      method: "Post",
-      headers: {
-        Accept: "application/json, text/plain, */*",
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        email: email,
-        password: password,
-      }),
+    await fetch(USER_API_ENDPOINT + "/register", {
+        method: "Post",
+        headers: {
+            Accept: "application/json, text/plain, */*",
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            email: email,
+            password: password,
+            name: "Username placeholder",           // To be changed in the future
+            major: "User's major placeholder",      // To be changed in the future
+            course : "",                            // To be changed in the future
+            role: "User role placeholder"           // To be changed in the future
```

Comment:
I think for this by default every new registered user will be "student" right? Last time we decided to only promote a user to "tutor" if he specifically requests it.

Reply:
Sure, I'll change this to "student" instead as a default role. However we might need to think an alternative way to handle this, as registering a user as "student" then promote to "tutor" sound breaking OOPs to me (eventhough we are not necessarily following it)